dev-docs/RFCs/v8.0/coordinate-spaces-rfc.md

```diff
@@ -0,0 +1,40 @@
+
+## Overview
+
+deck.gl's [coordinate systems](https://deck.gl/#/documentation/developer-guide/coordinate-systems) include the following:
+- World space (model space in standard 3D) - Whatever space the data is defined in, often longitude/latitude or meter offsets
+- Common space (world space in standard 3D) - Primary shared space into which world coordinates are transformed, measured in mercator pixels. +x points to the right, +y points down (following the web mercator convention), and +z points above the map surface. Note that unlike the other 3D spaces in deck, this makes common space a left-handed coordinate system. This space also includes the mercator zoom, so it scales up as the zoom increases.
+- View space - [Based on mapbox](https://deck.gl/#/documentation/developer-guide/coordinate-systems). Functions almost as a standard view space would, with the camera translated to the origin and the axes rotated so that +x points to the right, +y up and -z into the scene. There is, however, an odd [scaling of the z axis](https://github.com/uber-common/viewport-mercator-project/blob/master/src/web-mercator-utils.js#L195) that makes this space non-isotropic.
+- Clip space - The same as standard clip space (as required by the gl). +x is right, +y is up, +z is into the scene, coordinates from (-1, -1, -1) to (1, 1, 1).
+
+The key problems with these system are the following:
+- Left-handed common space means it must be treated as a special case compared to the other spaces, e.g. cross products must be reversed.
+- Scaled z-axis in view space means common space and view space aren't using the same units. This makes it difficult to map distances between the two or reason about the frustum in common space.
+- Zoom applied to common space can be unintuitive. E.g. defining a position in common space requires taking the zoom level into account. In most systems, zoom is considered a camera (view or projection) transform.
+
+
+## Implementation
+
+The problem of a left-handed common space is easily remedied by modifying the relevant parts of the mercator projection and view matrix calculations. This simply amounts to changing the mercator projection in [project.glsl.js](https://github.com/uber/deck.gl/blob/master/modules/core/src/shaderlib/project/project.glsl.js#L99-L102) to:
+```c
+return vec2(
+  radians(x) + PI,
+  PI + log(tan_fp32(PI * 0.25 + radians(lnglat.y) * 0.5)) // Note it's now `PI + ...`
+);
+```
+and removing -y scaling in the initialization of the view matrix in [viewport.js](https://github.com/uber/deck.gl/blob/master/modules/core/src/viewports/viewport.js#L354).
+
+The issue of scaling the z-axis is also easily resolved by removing [that scaling](https://github.com/uber-common/viewport-mercator-project/blob/master/src/web-mercator-utils.js#L195) from the view matrix calculation and adjusting the [near and far planes](https://github.com/uber-common/viewport-mercator-project/blob/master/src/web-mercator-utils.js#L236-L237) accordingly.
```

Comment:
The odd scaling for the `MapView` case is inherited from how mapbox creates their view matrix. I recall trying to refactor this but got side effects. 

Some subtle like changing the size of point cloud points projected at "ground distance" (the current matrix setup seems to ensure the scale at ground distance is `1`.)

If we can solve all of those (if necessary, using additional code in other places) it would be great to create that matrix in a more standard way.

Reply:
I was able to remove this without any obvious side effects, but I'm not following what the problems you faced were. Let's discuss in our 1:1 this week.